### ISSUE TITLE
Support staging micropub servers

### ISSUE THREAD

**Issue by jamesgoca** [aaronpk/Quill#132]
I have been testing my micropub server live using Quill now that I'm almost done. One issue I have encountered is that I cannot set a staging URL to which I can send test requests. I have had to change the micropub URL on my website to and from the staging URL during testing.

This may be somewhat of an edge case but it has been a major pain point while testing Quill. Some proposed ideas:

- Allow users to specify a test URL as a header on their website to which they can send requests
- Allow Quill to refresh a site without having to log out and authenticate from scratch

I am unsure whether these have broader interest so I'd like to hear from the community.

**Comment by jamietanna** [comment]
[My implementation](https://www.jvt.me/posts/2019/12/26/micropub-staging-server/) of a staging server is similar to yours - I have a separate identity + Micropub server for this purpose.

It appears that others in the community may be looking at the <code>[draft</code>](https://github.com/indieweb/micropub-extensions/issues/24) scope instead for this purpose

(Originally published at: https://www.jvt.me/mf2/2020/09/dstzp/)

**Comment by jamietanna** [comment]
Regarding logging in and logging out - I'd thoroughly recommend [Firefox multi-account containers](https://addons.mozilla.org/en-US/firefox/addon/multi-account-containers/) for this - I use it and it makes it super easy to flip between accounts and servers

(Originally published at: https://www.jvt.me/mf2/2020/09/c1qwk/)

**Comment by jamesgoca** [comment]
Thank you for the recommendation! I did not know that I could just authenticate in different environments. Would that solve my initial problem? 

 am no longer actively working on my micropub client but I think there will be others who have a similar issue. Even if I authenticate in a different session, I would still need to manually change the header tags on my site with my staging URL.

A potential solution could be to have a `<meta>` tag that lets a user specify a staging URL. Let me know if I am missing something obvious :)

**Comment by jamietanna** [comment]
Yes that's correct - if you wanted to continue using your primary identity URL, you'd need to update <code>&lt;meta&gt;</code> tags / <code>Link</code> HTTP headers.

For that reason, and to reduce risk of accidentally pushing something to my live site, when testing my staging infrastructure, I chose a separate identity, <code>www.staging.jvt.me</code> which simply hosts <code>&lt;meta&gt;</code> links to my staging server.

(Originally published at: https://www.jvt.me/mf2/2020/09/btcur/)

**Comment by capjamesg** [comment]
Thanks for sharing your approach! It seems better to set up a staging site on which I could test first before setting up the main endpoint. I am no longer actively working on my micropub client but I think this documentation could be helpful for others.
